composer/trainer/trainer.py

```diff
@@ -2140,12 +2140,155 @@ def predict_batch_end(self, state: State, logger: Logger) -> None:
         return outputs
 
     def eval(
+        self,
+        eval_dataloader: Optional[Union[Iterable, DataSpec, Evaluator, Sequence[Evaluator]]] = None,
+        subset_num_batches: int = -1,
+        # deprecated arguments below
+        dataloader: Optional[Any] = None,
+        dataloader_label: Optional[Any] = None,
+        metrics: Optional[Any] = None,
+        log_level: Optional[Any] = None,
```

Comment:
```suggestion
        self,
        eval_dataloader: Optional[Union[Iterable, DataSpec, Evaluator, Sequence[Evaluator]]] = None,
        subset_num_batches: int = -1,
        # deprecated arguments below
        **kwargs,
```
How about we just set this to kwargs and then we check if any of the deprecated arguments are specified inside the kwargs? This way we don't have args in the signature that we don't want people to use, which would be confusing.

Reply:
Ah i like this better, thanks!